### PR TITLE
(SIMP-4897) Updated Puppet installation methods

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Updated README
* Changed acceptance tests to use `beaker:suites`
* Removed all Puppet 5+ mappings and updated the install method to
  figure out what to use based on the available gems so that everything
  is now consistent

### CODE REVIEW

File: lib/simp/beaker_helpers.rb [hunk 1]
```diff
@@ -693,40 +693,88 @@ def pluginsync_on( suts = hosts )
...
   end
 
 
   # Replacement for `install_puppet` in spec_helper_acceptance.rb
   def install_puppet
+    # The first match is internal Beaker and the second is legacy SIMP
+    puppet_agent_version = ENV['PUPPET_INSTALL_VERSION'] || ENV['PUPPET_VERSION']
+
+    # In case Beaker needs this down the line
+    ENV['PUPPET_INSTALL_VERSION'] = puppet_agent_version
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers.rb]: why is the ENV update done here and not after latest_puppet_agent_version_for() is called on line 773?

File: files/puppet-agent-versions.yaml [hunk 2]
```diff
@@ -4,17 +4,6 @@
...
-  '5.3.4': '5.3.4'
-  '5.3.3': '5.3.3'
-  '5.3.2': '5.3.2'
-  '5.3.1': '5.3.1'
-  '5.3.0': '5.3.0'
-  '5.2.0': '5.2.0'
-  '5.1.0': '5.1.0'
-  '5.0.1': '5.0.1'
-  '5.0.0': '5.0.0'
   '4.10.7': '1.10.7'
```
Comment (jeefberkey) [inline files/puppet-agent-versions.yaml]: I think there's up to a 4.10.11 now, those should be added.

File: files/puppet-agent-versions.yaml [hunk 3]
```diff
@@ -4,6 +4,10 @@
 #   - https://docs.puppet.com/puppet/latest/about_agent.html
 #
 version_mappings:
+  '4.10.11': '1.10.12'
```
Comment (jeefberkey) [inline files/puppet-agent-versions.yaml]: Puppet what are you doing :man_facepalming:

File: lib/simp/beaker_helpers.rb [hunk 4]
```diff
@@ -687,7 +687,7 @@ def pluginsync_on( suts = hosts )
   # Looks up latest `puppet-agent` version by the version of its `puppet` gem
   #
   # @param puppet_version [String] target Puppet gem version.  Works with
-  #   Gemfile comparison syntax (e.g., '4.0', '= 4.2', '~> 4.3.1')
+  #   Gemfile comparison syntax (e.g., '4.0', '= 4.2', '~> 4.3.1', '> 4.10, < 5.5')
```
Comment (jeefberkey) [inline lib/simp/beaker_helpers.rb]: @lnemsick-simp didn't you say this new range doesn't work as implied here?

**Pull request merged**